cumulusci/tasks/push/tests/test_push_tasks.py

```diff
@@ -0,0 +1,518 @@
+
+def test_report_push_status_error():
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.sf = mock.MagicMock()
+    task.push_report = mock.MagicMock()
+    task.sf.query_all.return_value = {"totalSize": 0, "records": []}
+    with pytest.raises(PushApiObjectNotFound):
+        task._report_push_status("0DV1R000000k9dEWAQ")
+
+
+def test_get_push_request_job_results(
+    package_push_job_success, package_push_job_failure, package_push_job_cancel
+):
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.sf = mock.MagicMock()
+    task.push_report = mock.MagicMock()
+    task.push_request = mock.MagicMock()
+    task.push_request.get_push_job_objs.return_value = [
+        package_push_job_success,
+        package_push_job_failure,
+        package_push_job_cancel,
+    ]
+    task._get_push_request_job_results()
+    task.push_request.get_push_job_objs.assert_called_once()
+
+
+def test_schedule_push_org_query_get_org_error():
+    task = create_task(
+        SchedulePushOrgQuery,
+        options={
+            "orgs": ORG,
+            "version": VERSION,
+            "namespace": NAMESPACE,
+            "start_time": None,
+            "batch_size": "200",
+            "min_version": "1.1",
+        },
+    )
+    task.push = mock.MagicMock()
+    task.push_api = mock.MagicMock()
+    task.sf = mock.Mock()
+    task.push_api.return_query_records = mock.MagicMock()
+    task.push_api.return_query_records.return_value = {"totalSize": "1"}
+    task.sf.query_all.return_value = PACKAGE_OBJS
+    with pytest.raises(ValueError):
+        assert task._get_orgs() == [NAME]
+
+
+def test_schedule_push_org_query_get_org():
+    task = create_task(
+        SchedulePushOrgQuery,
+        options={
+            "orgs": ORG,
+            "version": VERSION,
+            "namespace": NAMESPACE,
+            "start_time": None,
+            "batch_size": "200",
+            "subscriber_where": "OrgType = 'Sandbox'",
+        },
+    )
+    task.push = mock.MagicMock()
+    task.push_api = mock.MagicMock()
+    task.sf = mock.Mock()
+    task.sf.query_all.return_value = PACKAGE_OBJ_SUBSCRIBER
+    assert task._get_orgs() == ["bar"]
+
+
+def test_schedule_push_org_list_run_task_with_time_assertion(org_file):
+    task = create_task(
+        SchedulePushOrgList,
+        options={
+            "orgs": ORG_FILE,
+            "version": VERSION,
+            "start_time": datetime.datetime(2021, 8, 20, 3, 55).strftime(
+                "%Y-%m-%dT%H:%M"
+            ),
+            "namespace": NAMESPACE,
+        },
+    )
+    task.push = mock.MagicMock()
+    task.sf = mock.MagicMock()
+    task.sf.query_all.return_value = PACKAGE_OBJS
+    task.push.create_push_request.return_value = (task.sf.query_all.return_value, 2)
+    task._run_task()
+    task.push.create_push_request.assert_called_once_with(
+        task.push.get_package_objs()
+        .__getitem__()
+        .get_package_version_objs()
+        .__getitem__(),
```

Comment:
@jofsky is this just a hack to get the same mock objects that were used for the actual call? Because it looks quite odd to be calling `__getitem__` without passing any argument. Since these are not the point of the assertion, you could pass unittest.mock.ANY instead: https://docs.python.org/3/library/unittest.mock.html#any